contribs/gmf/src/d3-lidar-profile/lidarprofileMeasure.js

```diff
@@ -0,0 +1,200 @@
+goog.provide('gmf.lidarProfile.measure');
+
+
+/**
+* Measure tool for the d3 chart
+* @constructor
+* @param {Object} options to be defined in gmfx
+* @param {gmf.lidarProfile} parent instance
+*/
+gmf.lidarProfile.measure = function(options, parent) {
+
+/**
+* @type {Object}
+* @private
+*/
+  this.options_ = options;
+
+  /**
+  * @type {gmfx.lidarMeasure}
+  * @private
+  */
+  this.profileMeasure_ = {
+    pStart: {
+      distance: undefined,
+      altitude: undefined,
+      color_packed: [],
+      coords: [],
+      intensity: undefined,
+      classification: undefined,
+      set: false
+    },
+    pEnd: {
+      distance: undefined,
+      altitude: undefined,
+      color_packed: [],
+      coords: [],
+      intensity: undefined,
+      classification: undefined,
+      set: false
+    }
+  };
+
+  /**
+  * @type {gmf.lidarProfile}
+  */
+  this.parent_ = parent;
+
+
+};
+
+/**
+* Clear the current measure
+* @export
+*/
+gmf.lidarProfile.measure.prototype.clearMeasure = function() {
+
+  this.profileMeasure_ = {
+    pStart: {
+      distance: undefined,
+      altitude: undefined,
+      color_packed: [],
+      coords: [],
+      intensity: undefined,
+      classification: undefined,
+      set: false
+    },
+    pEnd: {
+      distance: undefined,
+      altitude: undefined,
+      color_packed: [],
+      coords: [],
+      intensity: undefined,
+      classification: undefined,
+      set: false
+    }
+  };
+
+  const svg = d3.select('svg#profileSVG');
+  svg.selectAll('#text_m').remove();
+  svg.selectAll('#start_m').remove();
+  svg.selectAll('#end_m').remove();
+  svg.selectAll('#line_m').remove();
+
+  d3.select('#height_measure').html('');
+  d3.select('svg#profileSVG').on('click', null);
+
+  d3.select('svg#profileSVG').style('cursor', 'default');
+
+
+};
+
+/**
+* Activate the measure tool
+* @export
+*/
+gmf.lidarProfile.measure.prototype.setMeasureActive = function() {
+
+  const that = this;
+
+  d3.select('svg#profileSVG').style('cursor', 'pointer');
+
+  function measureHeigt() {
```

Comment:
Please, move out this function (to `gmf.lidarProfile.measure.prototype.measureHeight`)

And please never do thing like `const that = this;`, use my_function.bind(this) and use `this` inside

Reply:
Fixed